### DISCUSSION TITLE
Installing haxo-rs

### DISCUSSION THREAD

**Discussion by raychmond** [cardonabits/haxo-hw#14] [Q&A] [answered]
@jcard0na 
Hi Javier,

Good news! I've got the sound.

Here is the story.
First, I didn't see that fluidsynth warning after I copied the compiled binary file to /usr/local/bin and ran sudo /usr/local/bin/haxo001

Probably, the sound have already come out, but I don't know yet because I kept using the hat's amp.

I notice that you have a script install.sh and haxo.service. I ran it, but it doesn't help. Is it necessary to run a service? When I check the service status by running "sudo service --status-all" haxo service doesn't show up.

I saw you have a TenorSax2.sf2 in the service file. But I don't have that file in the /usr/share/sounds/sf2. So I downloaded someone's tenor sax sf2 file and copied there. But it doesn't help either.

At last, when I almost gave up. I switched the audio to the raspberry pi built-in audio. I heard the sound.  So, I guess I just need the last step to work, the amp's audio driver.

BTW, I've shipped the PCBs to you and Don. I will email you and Don about the cost.

I agree that I've put too many questions in another discussion. So I opened this new one here about the installation of haxo-rs.

Thanks!

**Comment by jcard0na** [comment, marked as answer]
Hi @raychmond,

I realized I never documented how to switch to the onboard audio codec.  I just did:  https://github.com/jcard0na/haxo-rs/blob/main/README.md#sound-card

With that configuration, you should see the soundcard as...
```
$ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: MAX98357A [MAX98357A], device 0: bcm2835-i2s-HiFi HiFi-0 [bcm2835-i2s-HiFi HiFi-0]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
```

Let me know if that worked.

**Reply by raychmond** [reply]
It works perfectly. The sound from the soundcard max98357a is amplified. It takes less efforts to play the haxo. I love it.

**Comment by jcard0na** [comment]
Regarding the service... no it is not required to run `haxo001` as a service, it is just a convenience.
The `install.sh` script will just copy the service definition file to `haxo/etc/systemd/system`.  The convenience of a service is that it can be easily configured to run on boot, but there are other ways to achieve the same.

By the way, do not forget to configure the overlay.  Otherwise you risk bricking the SD card right before demoing the haxophone.  It's easy to recover, but a bit embarrassing... (true story).

**Reply by raychmond** [reply]
Noted. Thank you for reminding me Javier!
